schema/manifestlist_test.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/opencontainers/image-spec/schema"
+)
+
+func TestManifestList(t *testing.T) {
+	for i, tt := range []struct {
+		strict       bool
+		manifestList string
+		fail         bool
+	}{
+		// valid manifest list accepting customized media types
+		{
+			strict: false,
+			manifestList: `
+{
+  "schemaVersion": 2,
+  "mediaType": "application/customized+json",
+  "manifests": [
+    {
+      "mediaType": "application/customized+json",
+      "size": 7143,
+      "digest": "sha256:e692418e4cbaf90ca69d05a66403747baa33ee08806650b51fab815ad7fc331f",
+      "platform": {
+        "architecture": "ppc64le",
+        "os": "linux"
+      }
+    }
+  ]
+}
+`,
+			fail: false,
+		},
+
+		// expected failure strictly: manifestList.mediaType is unknown
+		{
+			strict: true,
+			manifestList: `
+{
+  "schemaVersion": 2,
+  "mediaType": "application/customized+json",
```

Comment:
This will fail regardless of strict (since #404).  I'm fine leaving a test for that in place (with `fail: true`), but the test should use `strict: false` to show that the failure is due to you not passing a manifest-list, and not due to particular validation strictness.